docs/book/src/developer/providers/contracts/infra-cluster.md

```diff
@@ -308,6 +308,8 @@ If a condition with type `Ready` exist, such condition will be mirrored in Clust
 Please note that the `Ready` condition is expected to surface the status of the InfraCluster during its own entire lifecycle,
 including initial provisioning, the final deletion process, and the period in between these two moments.
 
+Provider implementers SHOULD implement `Status.Conditions[Paused]` to report if the cluster is paused. It should check if 'spec.paused' is set on the cluster, and for the paused annotation. Implementing this condition is currently optional, but we may revisit this in the future.
```

Comment:
I think I would add a dedicated, optional rule for pause stating that
- providers SHOULD implement the pause behaviour for every object with a reconciliation loop (eg. pause machines when the cluster or the object is paused; pause does not makes sense for templates)
- if implementing the pause behaviour, providers SHOULD surface the pause status for an object with a condition

Also, let's be specific and name the annotation that trigger pause at object level

Reply:
Ok, I think I've updated as you've asked! PTAL :)